integrations/pcap_analyzers/check_pcap.py

```diff
@@ -0,0 +1,131 @@
+# reference
+# https://github.com/OISF/suricata/blob/master/python/suricata/sc/suricatasc.py
+import argparse
+import json
+import logging
+import os
+import select
+import time
+from socket import AF_UNIX, socket
+
+LOG_NAME = "check_pcap"
+
+# get flask-shell2http logger instance
+logger = logging.getLogger("flask_shell2http")
+# logger config
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+log_level = os.getenv("LOG_LEVEL", logging.INFO)
+log_path = os.getenv("LOG_PATH", f"/var/log/intel_owl/{LOG_NAME}")
+# create new file handlers, files are created if doesn't already exists
+fh = logging.FileHandler(f"{log_path}/{LOG_NAME}.log")
+fh.setFormatter(formatter)
+fh.setLevel(log_level)
+fh_err = logging.FileHandler(f"{log_path}/{LOG_NAME}_errors.log")
+fh_err.setFormatter(formatter)
+fh_err.setLevel(logging.ERROR)
+# add the handlers to the logger
+logger.addHandler(fh)
+logger.addHandler(fh_err)
+logger.setLevel(log_level)
+
+
+class Suricata:
+    def __init__(self):
+        try:
+            # connect to Suricata Socket
+            self.s = socket(AF_UNIX)
+            parser = argparse.ArgumentParser()
+            parser.add_argument("filename", type=str)
+            parser.add_argument("file_md5", type=str)
+            parser.add_argument("job_id", type=str)
+            parser.add_argument("--reload_rules", action="store_true")
+            args = parser.parse_args()
+            logger.info(
+                f"received args: filename: {args.filename}, md5: {args.file_md5},"
+                f" reload_rules {args.reload_rules}, job_id: {args.job_id}"
+            )
+            self.filename = str(args.filename)
+            self.job_id = args.job_id
+            self.md5 = args.file_md5
+            self.reload_rules = args.reload_rules
+            self.analysis_dir = f"/tmp/eve_{self.job_id}"
+            self.eve_file = self.analysis_dir + "/eve.json"
+            self.check_pcap()
+        except Exception as e:
+            logger.exception(e)
```

Comment:
Do we need to try/catch all this block + generic exception? Can't we limit the scope?

Reply:
if you don't do that, unexpected exceptions won't be logged and you'll never know that something has broken